### PR TITLE
Workflows: adjust eamxx-v1 tests folder names

### PR DESCRIPTION
**Pull request by bartgol**
Ensure that, when we clean up at the end, we don't accidentally remove the folder
of another version of the same test, currently running as part of another PR CI testing.

[BFB]

---

As an example, I think these two ran together:

- [this](https://github.com/E3SM-Project/E3SM/actions/runs/16974255810/job/48118900377?pr=7605) test failed the cleanup phase because a folder was misteriously not empty
- [this](https://github.com/E3SM-Project/E3SM/actions/runs/16971712702/attempts/1?pr=7549) test failed b/c of missing files in the test folder

What I think happened is this:
- test 1 started to cleanup ALL folders whose name matched the pattern
- it gathered the list of file to delete, and started deleting them. This list included files in the folder for the second test
- by the time it deleted the files, the 2nd test had created more files in the test folder, so test 1 could not remove the folder, as it was now no longer empty, so it failed
- test 2 found a corrupted test folder with missing files, so it failed.

This PR uses the sha of the tested branch as part of the test id, which allows us to remove only the correct folder in the cleanup phase, since we now know _exactly_ the full folder name (before, `create_test` was appending the id made with the test creation date and a sha of something).

### CODE REVIEW

**Comment by bartgol** [comment]
@jgfouca I think I am missing something regarding create_test. Take a look at the [last CI run](https://github.com/E3SM-Project/E3SM/actions/runs/16978830849/job/48134284482?pr=7610): the two tests that fails are I think due to create_test being monitoring "too many tests". I wonder if the problem is in the `-t` flag: by giving the same test id to all 5 builds, there is just one `cs.status.<the_sha>` for all tests, which gets overwritten every time a new test is created?

Assuming that's right, what's the best course of action here? Perhaps add also `--output-root <TEST_NAME>`, so that all the stuff for test XYZ is in `/projects/e3sm/scratch/XYZ/`, without clash?

Edit: I tried to also add `--output-root <TEST_NAME>`, so that all the files generated by create_test from different jobs will be in separate folders.

**Comment by jgfouca** [comment]
@bartgol , the idea we have used in the nightlies/jenkins for years has been to add a timestamp to the test_id to ensure that it is unique for each run. If you can run shell commands in the yaml, you can get a timestamp similar to the CIME one via the command `date +%y%m%d_%H%M%S`.

**Comment by mahf708** [comment]
`-t ${{ github.sha }}` will get rid of the `date +%y%m%d_%H%M%S` in tests names. 

I don't see the need for `-t ${{ sha }}` but if you really want to keep something like it, you can try to append `date +%y%m%d_%H%M%S` to the tag (`-t ${{ sha }}_${{ mydate }}` or something --- likely something like mydate is already in the env)

**Comment by bartgol** [comment]
@mahf708 the reason for `-t ${{ github.sha }}` is that we know that programmatically, while the timestamp that create_test uses is whatever the time/date was when create_test ran. I think using github.sha is already uniquely defining a folder (there should be no _concurrent_ jobs using the same github sha), so I think that's already enough. We don't really care about the date/time in the scratch folder created by create_test. Also, if I add a time/date stamp, I also have to save it to an env var, so I can use it later in the upload artifact phase. Given all of this, I figured using just github.sha was enough, no?

@jgfouca in order to retrieve the stamp, I would have to first cd into the test folder, and therefore know exactly the name, and therefore know the stamp (a catch22 situation). The problem is that 2 occurrences of the same test _may_ be running at the same time (if 2 PRs are concurrently tested), in which case I don't know which one if the right folder for this test.

Unless there's a way to tell create_test to return the test id used (or maybe set it in some env var), I think the only solution is to make create_test use a pre-defined id. One way is to ping the current date/time, save it as an env var, and use that for the create_test phase and upload artifact phase. But I figured github.sha worked just as well.

**Comment by jgfouca** [comment]
@bartgol , 

> Unless there's a way to tell create_test to return the test id used (or maybe set it in some env var), I think the only solution is to make create_test use a pre-defined id.

That's what I meant. Just use the `date` command to generate a test id and pass it to create_test.

**Comment by bartgol** [comment]
> @bartgol ,
> 
> > Unless there's a way to tell create_test to return the test id used (or maybe set it in some env var), I think the only solution is to make create_test use a pre-defined id.
> 
> That's what I meant. Just use the `date` command to generate a test id and pass it to create_test.

Ok, I agree, that works. But is there a reason to do that instead of using github.sha?

**Comment by bartgol** [comment]
Ah, wait, with github.sha, the testreporter is the same for all tests, so it gets confused (as [this](https://github.com/E3SM-Project/E3SM/actions/runs/16978830849/job/48134284484?pr=7610) fail showed). Adding the test name as output root did not work, as it failed to setup the cs.xyz files I think (for some reason).

Ok, I will switch to sha+timestamp (yes, the odds of 2 concurrent jobs with same timestamp are ridicoulously small,but why not?).

**Comment by bartgol** [comment]
@mahf708 @jgfouca I think I managed to get a deterministic testid that does not clash with any other job. I also shortened the short name of the v1 tests, so it takes less space when printed in the actions summary.

File: .github/workflows/eamxx-v1-testing.yml [hunk 1]
```diff
@@ -103,24 +108,27 @@ jobs:
             echo "Unsupported Linux distribution"
             exit 1
           fi
+      - name: Set test id
+        run: |
+          GITHUB_SHA=${{ github.sha }}
+          SHORT_SHA=${GITHUB_SHA::7}
+          TESTID=$SHORT_SHA-${{ matrix.test.id }}-$(date -u +"%y%m%d-%H%M%S")
```
Comment (mahf708) [inline .github/workflows/eamxx-v1-testing.yml:115]: hmmmmm why was the matrix.test.id necessary?

Comment (mahf708) [reply .github/workflows/eamxx-v1-testing.yml:115]: (obvz feel free to ignore; I think what you have is fine; I'm just curious what corner case you're taking care with test.id)

Comment (bartgol) [reply .github/workflows/eamxx-v1-testing.yml:115]: The problem is the cs.status.TESTID file, that CIME use to monitor the test while it's running (so it can later report the pass/fail status). Since the TESTID was the same for all tests (which run in different jobs), and since the scratch folder is mounted (and therefore shared across concurrent jobs), each job was basically using the same cs.status.xyz file in the scratch folder.

In [this](https://github.com/E3SM-Project/E3SM/actions/runs/17106504469/job/48516709737?pr=7610) CI run you can see in the log the confusion:
```
Waiting for tests to finish
PEND ERS_Ln9.ne4_ne4.F2000-SCREAMv1-AQP1.ghci-snl-cpu_gnu.eamxx-output-preset-2 (phase MODEL_BUILD unexpectedly left in PEND)
    Case dir: /projects/e3sm/scratch/ERS_Ln9.ne4_ne4.F2000-SCREAMv1-AQP1.ghci-snl-cpu_gnu.eamxx-output-preset-2.C.136f927221b53dbbc07ddc6c5b12f63434ea0da0-250820-180745
PASS REP_D_Ln5.ne4pg2_oQU480.F2010-EAMxx-MAM4xx.ghci-snl-cpu_gnu RUN
    Case dir: /projects/e3sm/scratch/REP_D_Ln5.ne4pg2_oQU480.F2010-EAMxx-MAM4xx.ghci-snl-cpu_gnu.C.136f927221b53dbbc07ddc6c5b12f63434ea0da0-250820-180745
```
This run was for the REP_D test, but ended up monitoring also the ERS, b/c of the clash between cs.status.xyz files (I think). @jgfouca may disprove my theory though.

I think we could just use sha+matrix.test.id as TESTID, but I figured a timestamp is still ok to have.

Comment (jgfouca) [reply .github/workflows/eamxx-v1-testing.yml:115]: @bartgol , that is not quite correct. cs.status.XXX is a user tool for getting a quick dump of the statuses of a run. The internal mechanics of CIME use wait_for_tests, which looks at `$outputroot/*$testid/TestStatus`

Comment (mahf708) [reply .github/workflows/eamxx-v1-testing.yml:115]: I'm kinda weirded out by all of this tbh

- the idea of using containers is to isolate stuff so that when you run something (say `ERS_Ln9.ne4_ne4.F2000-SCREAMv1-AQP1.ghci-snl-cpu_gnu.eamxx-output-preset-2`) you can't really see other tests and what they are doing. I suspect the issue here is that you're exposing the disk for these tests, which is not harmful, but still not as good. I would let each instance of the container only see the baselines dir for read-only access, and then have them write their test stuff in a place like /tmp that vanishes when the container instance vanishes
- I also have never seen this situation where multiple tests are waiting on each other; my understanding aligns with what Jim is saying; something somewhere must be acting up?

Comment (bartgol) [reply .github/workflows/eamxx-v1-testing.yml:115]: Yeah, I think we could just avoid mounting the scratch folder, and just use local space in the container. I think initially I wanted to keep the scratch folder for failed tests maybe? I forgot. And usually the uploaded logs/nc files are enough to diagnose issues anyways. I think I will go and modify the testing so that we don't mount the scratch folder.

Re: tests waiting on each other. Ok, not cs.status.xyz, but the issue is basically the same: for all tests, CIME looks at `$outputroot/*$testid/TestStatus`, and since the test shared the outputroot (since scratch is mounted), if testid only contains sha+date, two jobs of the same workflow that kick in simultaneously would look for the same TestStatus (as there's no trace of the actual test name in `$outputroot/*$testid/TestStatus`). Anyhow, all of this would go away once we make the scratch an internal folder in the container.

**Comment by bartgol** [comment]
I'm going to close this. I will open a new PR that cleans up the workflow (removing the cleanup step) once the containers have been updated to NOT mount /projects/e3sm/scratch
